src/global_helpers.R

```diff
@@ -976,3 +976,125 @@ fname_from_fpath <- function(paths, include_fext = TRUE){
 
     return(fnames)
 }
+
+#still in progress 
+delineate_watershed <- function(lat, long) {
+    
+    site <- tibble(x = lat,
+                     y = long) %>%
+        sf::st_as_sf(coords = c("y", "x"), crs = 4269) %>%
+        sf::st_transform(102008)
+    
+    start_comid <- nhdplusTools::discover_nhdplus_id(sf::st_sfc(
+        sf::st_point(c(long, lat)), crs = 4269)) 
+    
+    flowline <- nhdplusTools::navigate_nldi(list(featureSource = "comid",
+                                                 featureID = start_comid),
+                                            mode = "upstreamTributaries",
+                                            data_source = "")
+    
+    subset_file <- tempfile(fileext = ".gpkg")
+    
+    subset <- nhdplusTools::subset_nhdplus(comids = flowline$nhdplus_comid,
+                                           output_file = subset_file,
+                                           nhdplus_data = "download",
+                                           return_data = TRUE)
+    
+    flowlines <- subset$NHDFlowline_Network %>%
+        sf::st_transform(102008)
+    
+    catchments <- subset$CatchmentSP %>%
+        sf::st_transform(102008)
+    
+    upstream <- nhdplusTools::get_UT(flowlines, start_comid)
+    
+    watershed <- catchments %>%
```

Comment:
your spatial skills are 1337. looking forward to stepping through this function!